src/AnnotatedCode.c

```diff
@@ -73,3 +101,63 @@ R_API void r_annotated_code_print_json(RAnnotatedCode *code)
 	r_cons_printf ("%s\n", pj_string (pj));
 	pj_free (pj);
 }
+
+R_API void r_annotated_code_print_with_syntax_highlighting(RAnnotatedCode* code, RCore *core)
+{
+    if (code->annotations.len == 0) {
+        r_cons_printf("%s\n", code->code);
+        return;
+    }
+
+
+    size_t cur = 0;
+    size_t len = strlen(code->code);
+
+	RCodeAnnotation *annotation;
+	r_vector_foreach (&code->annotations, annotation) {
+        if (annotation->type != R_CODE_ANNOTATION_TYPE_SYNTAX_HIGHLIGHT)
+            continue;
+
+        // (1/3)
+        // now we have a syntax highlighting annotation.
+        // pick a suitable color for it.
+#define PALETE(x) (core->cons && core->cons->context->pal.x)? core->cons->context->pal.x 
+        const char* color = Color_RESET;
+        switch (annotation->syntax_highlight.type) {
+        case R_SYNTAX_HIGHLIGHT_TYPE_COMMENT:
+            color = PALETE(comment): Color_WHITE;
+            break;
+        case R_SYNTAX_HIGHLIGHT_TYPE_KEYWORD:
+            color = PALETE(pop): Color_MAGENTA;
+            break;
+        case R_SYNTAX_HIGHLIGHT_TYPE_DATATYPE:
+            color = PALETE(func_var_type): Color_BLUE;
+            break;
+        case R_SYNTAX_HIGHLIGHT_TYPE_FUNCTION_NAME:
+            color = PALETE(fname): Color_RED;
+            break;
+        case R_SYNTAX_HIGHLIGHT_TYPE_CONSTANT_VARIABLE:
+            color = PALETE(num): Color_YELLOW;
+        default:
+            break;
+        }
+
+        // (2/3)
+        // the chunk before the syntax highlighting annotation should not be colored
+        for (; cur < annotation->start && cur < len; cur++)
```

Comment:
it's a good practice always to use curly braces, even with a single line condition/loop
https://embeddedgurus.com/barr-code/2014/03/apples-gotofail-ssl-security-bug-was-easily-preventable/